File: b762(eng_mon)/Python/eng_mon.py
from sys import stdin, stdout
while True:
    num_of_reports = stdin.readline().strip('\n')
    if not num_of_reports: break
    num_of_reports = int(num_of_reports)
    num_of_kills, num_of_dies, num_of_assists, conti_kills = 0, 0, 0, 0
    while num_of_reports:
        report = stdin.readline()
        if report[0] == 'D':
            num_of_dies += 1
            if conti_kills < 3: stdout.write('You have been slained.\n')
            else: stdout.write('SHUTDOWN.\n')
            conti_kills = 0
        elif report[4] == 'A': num_of_assists += 1
        else:
            num_of_kills += 1
            conti_kills += 1
            if conti_kills < 3: stdout.write('You have slain an enemie.\n')
            elif conti_kills == 3: stdout.write('KILLING SPREE!\n')
            elif conti_kills == 4: stdout.write('RAMPAGE~\n')
            elif conti_kills == 5: stdout.write('UNSTOPPABLE!\n')
            elif conti_kills == 6: stdout.write('DOMINATING!\n')
            elif conti_kills == 7: stdout.write('GUALIKE!\n')
            else: stdout.write('LEGENDARY!\n')
        num_of_reports -= 1
    stdout.write(f'{num_of_kills}/{num_of_dies}/{num_of_assists}\n')

File: b762(eng_mon)/C++/eng_mon.cpp
#include <iostream>

int main(int argc, char const *argv[]) {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(0);
  int num_of_reports;
  while (std::cin >> num_of_reports) {
    std::string report;
    int num_of_dies = 0, num_of_kills = 0, num_of_assists = 0, conti_kills = 0;
    while (num_of_reports--) {
      std::cin >> report;
      if (report[0] == 'D') {
        num_of_dies++;
        std::cout << (conti_kills < 3? "You have been slained.": "SHUTDOWN.") << '\n';
        conti_kills = 0;
      }
      else if (report[4] == 'A') num_of_assists++;
      else {
        num_of_kills++;
        if (++conti_kills < 3) {
          std::cout << "You have slain an enemie." << '\n';
          continue;
        }
        switch (conti_kills) {
          case 3: std::cout << "KILLING SPREE!" << '\n'; break;
          case 4: std::cout << "RAMPAGE~" << '\n'; break;
          case 5: std::cout << "UNSTOPPABLE!" << '\n'; break;
          case 6: std::cout << "DOMINATING!" << '\n'; break;
          case 7: std::cout << "GUALIKE!" << '\n'; break;
          default: std::cout << "LEGENDARY!" << '\n'; break;
        }
      }
    }
    std::cout << num_of_kills  << '/' << num_of_dies << '/' << num_of_assists << '\n';
  }
  return 0;
}
